test/governance/candidacy.users.test.ts

```diff
@@ -0,0 +1,238 @@
+
+    await candidate.runForCouncil();
+    await waitForNBlocks(2);
+
+    const termZero = getLastBlocksElectionData();
+
+    expect(flattenArray(termZero.Candidates)).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+  });
+
+  test("Candidate can renounce candidacy", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    await candidate.runForCouncil();
+    await waitForNBlocks(2);
+
+    await candidate.renounceCandidacy({Candidate: 1});
+
+    await waitForNBlocks(2);
+
+    const termZero = getLastBlocksElectionData();
+
+    expect(flattenArray(termZero.Candidates)).not.toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+  });
+
+  test("Council Member can renounce candidacy", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    // Gonzalo runs for Council in Term 0
+    // Eddy votes for Gonzalo
+    await candidate.runForCouncil();
+    await voter.vote([candidate], stakeAmount);
+    await waitForNBlocks(2);
+
+    const termZero = getLastBlocksElectionData();
+
+    expect(flattenArray(termZero.Candidates)).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+
+    // Gonzalo wins the election and is council member!
+    // Gonzalo subsequently renounces candidacy in Term 1
+    await waitForNextTerm();
+
+    const termOne = getLastBlocksElectionData();
+    expect(flattenArray(termOne.Members)).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+
+    const candidateStatus = {Member: candidate.keyRingPair.address};
+    await candidate.renounceCandidacy(candidateStatus);
+
+    // Gonzalo is not automatically running for candidacy in Term 2
+    await waitForNextTerm();
+
+    const termTwo = getLastBlocksElectionData();
+
+    expect(flattenArray(termTwo.Candidates)).not.toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+  });
+
+  test("Council Member automatically runs for candidacy again", async () => {
+    const candidatesAddress = candidate.keyRingPair.address;
+
+    // Gonzalo runs for Council in Term 0
+    // Eddy votes for Gonzalo
+    await candidate.runForCouncil();
+    await voter.vote([candidate], stakeAmount);
+
+    await waitForNBlocks(2);
+
+    const termZero = getLastBlocksElectionData();
+    expect(flattenArray(termZero.Candidates)).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+
+    // Gonzalo wins the election and is a council member!
+    await waitForNextTerm();
+
+    const termOne = getLastBlocksElectionData();
+
+    expect(flattenArray(termOne.Members)).toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+
+    // Gonzalo is automatically running for candidacy in Term 2
+    await waitForNextTerm();
+
+    const termTwo = getLastBlocksElectionData();
+
+    expect(flattenArray(termTwo.Candidates)).not.toEqual(
+      expect.arrayContaining([candidatesAddress])
+    );
+  });
```

Comment:
Same as before, check that the user is still a member.

Reply:
He might not be! The only guarantee is that he's _running_ for election, not that he's been voted in again.